### DISCUSSION TITLE
Patching Gaps in the CIS Windows 11 Benchmark - Level 1 Windows Autopilot - odds+endpoints

### DISCUSSION THREAD

**Discussion by giscus** [ennnbeee/oddsandendpoints-web#3] [Comments]
# Patching Gaps in the CIS Windows 11 Benchmark - Level 1 Windows Autopilot

Updating the Center for Internet Security (CIS) benchmark build kit for Windows 11 Autopilot in Microsoft Intune for level 1 security settings.

https://memv.ennbee.uk/posts/windows-cis-patching-gaps-part2/

**Comment by Glazzanz** [comment]
This is very helpful, thanks!  Can you elaborate on "Disable Automatic Admin Logon" method?  I would have thought during the Autopilot process the OS is reported  as the correct version - so if Windows 11 it would report as something 10.0.22000.0000 and 10.0.22000.9999, and therefore the new policy would also apply, or is it that when the Autopilot process is running is reported as an older OS version?

**Reply by ennnbeee** [reply]
Yeah it'll report it's version of Windows, though the version in the applicability rule is only 10.0.**22000**.0000 to 10.0.**22000**.9999, which only covers all build versions of Windows 11 21H2. 

In support versions of Windows 11 have different builds numbers, e.g., 10.0.**23631**.xxxx so wouldn't be captured by the above rule as fall outside of the range of builds configured.

**Comment by Glazzanz** [comment]
Thanks.  I thought the “or later” would cover 10.0.23631.xxxx as well.

As an example, if we look at a setting that is in your *CIS (L1) Windows
Hello for Business - Windows 11 Intune 3.0.1*, that is specified in the CSP
as for *Windows 10 version 1511 [10.0.10586] and later*.

The setting in this example is:
*user_vendor_msft_passportforwork_{tenantid}_policies_usepassportforwork*

Working through a selection of other settings, there are very few settings
the explicitly state the are for newer versions of Windows 11 - hence my
confusion here.

**Reply by ennnbeee** [reply]
You make a good, point as there aren't many setting in the CSP that specify direct Windows 11 builds, so it's a little bit of an interpretation and cheat 😂.

I've thought of a couple of alternative ways of getting around it, so I'll update the post with the one that works when I test them.

**Comment by jimmywinberg** [comment]
This is such a great blog post, incredible work.
I can only say a big thank you for doing all this, such a time saver!

CIS should maybe rethink some if the settings you mentioned based on Intune to make the device useful or supply your workaround/fixes.
Have you mentioned this on the CIS forum?

Again this is the best post I ever seen regarding CIS and Intune.

**Reply by ennnbeee** [reply]
Thanks for the feedback mate, it's appreciated. 

I think the issue with the Windows build kit, is that it's designed to be tested first and is just provided "as-is", I had thought about mentioning this on the forum, though the settings they're recommended need to be implemented, and not amended. That's why it's important to not just use the build kit, but also consume the document, even if it is a lonnnnggg read.

**Reply by jimmywinberg** [reply]
I agree with you, still trying to get everything implemented and finding issues with:
CIS (L1) Device Lock & WHFB - Windows 11 Intune 3.0.1

Says conflicting on: Device Password History, Min Device Password Length, but not showing what policy it's conflicting with.
Assigned to All Devices.

If you know anything about above i would appreciate it.
I read version 1.0 now coming back to this and working my way thru it :)

**Reply by ennnbeee** [reply]
You might get conflict with that setting and a compliance policy requiring similar password requirements. I wouldn't recommend the compliance policy as this really only applies to local Windows accounts, so not particularly relevant for AD or Entra accounts.

**Reply by jimmywinberg** [reply]
Thank you, I will look into that, currently it looks like the conflict is pointing to itself, due to assigning it to All Devices.
I will look at removing the section in the compliancy policy and see if that resolves it.

Again you blog and your github is worth beyond gold :)

**Comment by CMartin249** [comment]
I don't understand why you would recommend "Turn on convenience PIN sign-in" be enabled. Convenience PIN and Windows Hello for Business are two separate things. Convenience PIN is a password stuffer while Windows Hello for Business is a passwordless authentication method. There is a policy to enable Windows Hello for Business that must be enabled which you did highlight. There are others that are needed as well that configure WHfB specific PIN rules and the deployment type to match the company's infrastructure.

**Reply by jimmywinberg** [reply]
I would agree with @CMartin249 It does not need to be turned on for Windows Hello to work.

This is more of a questions as when you assigned any Device Lock policy to a device you will have the "Other User" screen during deployment, Autopilot jumps out of deployment into the "Other User" screen and then you can sign-in and it continues the user policy parts.

Assign this to Users instead of device you will not see this affect and it will be a more smooth Autopilot deployment.
How do you approach this?

**Reply by ennnbeee** [reply]
Great shouts from both of you. The post has been updated to reflect the difference between Windows Hello and WHfB, also a note on assigning the Device Lock policy to users as opposed to devices (for this scenario that's fine, but for userless devices you're going to have to deal with some kind of pain)

**Reply by Glazzanz** [reply]
to clarify, you are saying that for a device that is used by a user the "CIS (L1) Device Lock & WHFB - Windows 11 Intune 3.0.1" should be assigned to a user, rather than a device?

**Comment by jimmywinberg** [comment]
Yes, you can do a test with Autopilot if you assign that policy to devices Auotpilot will jump out of provisioning to "Other User" screen and require a second login.

If you assign it to users this will not happen
________________________________

**Reply by Glazzanz** [reply]
Thanks for the clarification @jimmywinberg

**Comment by jimmywinberg** [comment]
New find: CIS (BL) BitLocker Misc - Windows 11 Intune 3.0.1 will also break Autopilot flow if assigned to devices.
Autopilot will jump out to Other User Screen, assigning it to user you will now have a smooth ride. I will keep posting these finds. You can of course live with the extra Other User sign in during provisioning if you want to.

**Comment by amirjs** [comment]
Great post thank you - Conscious  this is an older post but I am not sure I am entirely clear on this statement: "So once more to stripping out this setting into a separate profile, this time keeping the same configuration settings, but assigning this to all your corporate-owned devices, but excluding a group of Windows Autopilot devices configured for pre-provisioned deployments using rules based on the enrollmentProfileName associated with the device."

Do you mean that this separate profile will never apply to devices deployed by Autopilot (due to the exclusion) or do you mean that the config will apply only post pre-provisioning/enrollment? 
if it's the latter, what dynamic groups do you use to distiguish between a device pre enrollment and post enrollment? 

can you share the syntax of the dynamic group or filter used please?

**Reply by ennnbeee** [reply]
So the idea is that this setting only impacts autopilot devices, and only when using pre-prov...so assign the settings to All devices, excluding those autopilot ones. 

If you aren't doing pre-prov, then you don't need to exclude anything. If you are, there isn't an attribute available to group autopilot devices with pre-prov enabled, you'd have to group them by the enrolment profile name attribute.

If devices are excluded, then they will never get the setting.

The other way around something like this; is to assign the policy to a user group, as user settings aren't applied during pre-prov...so the setting will apply, just not during device/technician phase of autopilot, just user phase.

This is OK for user affinity and shared devices, not so good for userless kiosk devices, but you can't fix everything for every scenario.

**Reply by amirjs** [reply]
Thanks Nick. In my case all deployment profiles allow for pre-prov. I am aware of the workaround where we apply such settings to users rather than devices to avoid breaking autopilot pre-prov. I was just curious reading the above and wondered if there is another way of execluding the policy during pre-prov only.

In our case, we need to apply ALL CIS policies so I can't exclude autopilot deivces indefinitely. I will apply to users so the policy applies post pre-prov. Thanks!
